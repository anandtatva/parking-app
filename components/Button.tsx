import React from "react";
import { TouchableOpacity, Text, StyleSheet, TouchableOpacityProps, ViewStyle } from "react-native";
import Colors from "../constants/Colors";

interface ButtonProps extends TouchableOpacityProps {
    name: string;
    testId?: string;
    onPress: () => void;
    rootStyle?: ViewStyle;
}

export { ButtonProps }
export default function Button({ testId, onPress, name, rootStyle = {}, ...rest }: ButtonProps) {

    return (
        <TouchableOpacity testID={testId} onPress={onPress} style={[styles.container, rootStyle]} {...rest}>
            <Text style={styles.nameTxt}>{name}</Text>
        </TouchableOpacity>
    )

}

const styles = StyleSheet.create({
    container: {
        paddingHorizontal: 8,
        paddingVertical: 10,
        backgroundColor: Colors.dark.background,
    },
    nameTxt: {
        textAlign: "center",
        color: Colors.dark.text

    }
})