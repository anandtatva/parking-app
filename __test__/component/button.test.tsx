import { cleanup, render, waitFor, fireEvent } from "@testing-library/react-native";
import Button, { ButtonProps } from "../../components/Button";

describe("Button Component", () => {
    it("render correctly", () => {
        const props: ButtonProps = { onPress: jest.fn(), name: "Submit" }
        const tree = render(<Button  {...props} />).toJSON();
        expect(tree).toMatchSnapshot()
    })
    it("Button Must be called on press", () => {
        const props: ButtonProps = {
            testID: "Submit-Button",
            onPress: jest.fn(),
            name: "Submit"
        }
        const { getByTestId } = render(<Button  {...props} />);
        const btn = getByTestId("Submit-Button");
        fireEvent.press(btn)
        expect(props.onPress).toHaveBeenCalled()
    })
})