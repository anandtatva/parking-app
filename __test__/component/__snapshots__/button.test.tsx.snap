// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#000",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "textAlign": "center",
      }
    }
  >
    Submit
  </Text>
</View>
`;
